src/pages/Login/ResetPassword.js

```diff
@@ -1,51 +1,159 @@
 
 class ResetPassword extends Component {
     constructor(props) {
         super(props);
         this.state = {
             password: '',
             confirmPassword: '',
-            PasswordValidError: false
+            passwordMatchError: false,
+            PasswordValidError: false,
+            email: '',
+            updated: false,
+            isLoading: true,
+            error: false,
+            waitForRedirect: true,
+            redirectToLogin: false
         };
     }
 
+    async componentDidMount() {
+        const url = 'http://localhost:3001/users/resetpassword';
+        await axios
+            .get(url, {
+                params: {
+                    resetPasswordToken: this.props.match.params.token
+                }
+            })
+            .then(response => {
+                if (response.data.message === 'password reset link a-ok') {
+                    this.setState({
+                        email: response.data.email,
+                        isLoading: false
+                    });
+                } else {
+                    this.setState({
+                        isLoading: false,
+                        error: true
+                    });
+                }
+            })
+            .catch(error =>
+                alert('Oops. Something went wrong. Please try again')
+            );
+    }
+
+    componentDidUpdate() {
+        if (this.state.updated === true) {
+            this.first_delay = setTimeout(
+                () => this.setState({ waitForRedirect: false }),
+                3000
+            );
+        }
+        if (this.state.waitForRedirect === false) {
+            this.second_delay = setTimeout(
+                () => this.setState({ redirectToLogin: true }),
+                2000
+            );
+        }
+        if (this.state.redirectToLogin === true) {
+            this.props.history.push('/login');
+        }
+    }
+
+    componentWillUnmount() {
+        clearTimeout(this.first_delay);
+        clearTimeout(this.second_delay);
+    }
+
     handleChange = event => {
         this.setState({
             [event.target.name]: event.target.value
         });
     };
 
     handleSubmit = event => {
-        const { password, confirmPassword } = this.state;
+        const { email, password, confirmPassword } = this.state;
         event.preventDefault();
 
         let IsPasswordIdenticalVar = IsPasswordIdentical(
             password,
             confirmPassword
         );
         let isPasswordValidVar = IsPasswordValid(password);
-        if (IsPasswordIdenticalVar && isPasswordValidVar) {
-            (async () => {
-                const token = window.location.pathname;
-                const url = `http://localhost:3001/users${token}`;
-                await ky.post(url, { json: this.state }).then(res => {
-                    if (res.status === 200) {
-                        this.props.history.push('/login');
+
+        if (!IsPasswordIdenticalVar) {
```

Comment:
It seems you changed the logic of error message component
How it was originally?
1) On change, [passwordMessage.js](https://github.com/zero-to-mastery/mappypals/blob/dev/src/components/ErrorMessages/PasswordMessage/PasswordMessage.js) error checks the password and inform the user what's needed. On blur
1) [Helper.js](https://github.com/zero-to-mastery/mappypals/blob/master/src/components/helper.js) checks if the password is valid, depending on the result.  It either tries to change a password or not.

How it was changed:
1) The error message displays only after submit, not when the user left the input field.
2) Double checks, there's was already in PasswordMessage.js check if passwords are identical, however, it appears only after the user left confirm password field.
![password-don't-match](https://user-images.githubusercontent.com/38657595/60880403-46c46100-a24c-11e9-811a-dbc59e3543e7.jpg)

So the question is: Why you changed logic?

Reply:
I'm sorry, I didn't notice the change. The code has been updated now and works as expected.